src/main/java/gregtech/api/capability/impl/AbstractRecipeLogic.java

```diff
@@ -508,6 +530,51 @@ public void setParallelLimit(int amount) {
         parallelLimit = amount;
     }
 
+    /**
+     * Sets an EU/t discount to apply to a machine when running recipes.<br>
+     * This does NOT affect recipe lookup voltage, even if the discount drops it to a lower voltage tier.<br>
+     * This discount is applied pre-parallel/pre-overclock.
+     *
+     * @param discount The discount, must be greater than 0 and less than 1.
+     *                 If discount == 0.75, then the recipe will only require 75% of the listed power to run.
+     */
+    public void setEUDiscount(double discount) {
+        if (discount <= 0 || discount >= 1) {
+            GTLog.logger.warn("Cannot set EU discount for recipe logic to {}, discount must be 0<x<1", discount);
+            return;
+        }
+        euDiscount = discount;
+    }
+
+    /**
+     * @return the EU/t discount, or -1 if no discount.
+     */
+    public double getEUtDiscount() {
+        return euDiscount;
+    }
+
+    /**
+     * Sets a speed multiplier to apply to a machine when running recipes.<br>
+     * This discount is applied pre-parallel/pre-overclock.
+     *
+     * @param bonus The bonus, must be greater than 0.
+     *              If bonus == 0.2, then the recipe will be 20% of the normal duration.
+     */
+    public void setSpeedBonus(double bonus) {
```

Comment:
I don't see any reason why you shouldn't allow people to set this value above 1 in every single modpack and addon.

Reply:
Setting these above 1 would make them penalties, that wasn't the original purpose of this